Autopilot/PathManager/Inc/pathStateClasses.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include "pathManagerStateManager.hpp"
+#include "pathManager.hpp"
+#include "altimeter.hpp"
+#include "gps.hpp"
+#include "AutoSteer.hpp"
+#include "waypointManager.hpp"
+
+
+/***********************************************************************************************************************
+ * Code
+ **********************************************************************************************************************/
+
+class commsWithAttitude : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        sendToAttitude() {}
+        sendToAttitude(const sendToAttitude& other);
+        sendToAttitude& operator =(const sendToAttitude& other);
+};
+
+class getFromTelemetry : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        getFromTelemetry() {}
+        getFromTelemetry(const getFromTelemetry& other);
+        getFromTelemetry& operator =(const getFromTelemetry& other);
+};
+
+class getSensorData : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static AltimeterData_t *GetAltimeterOutput(void) {return &_altimeterdata;}
+        static GpsData_t *GetGPSOutput(void) {return &_gpsdata;}
+    private:
+        getSensorData() {}
+        getSensorData(const getSensorData& other);
+        getSensorData& operator =(const getSensorData& other);
+        static AltimeterData_t _altimeterdata; 
+        static GpsData_t _gpsdata; 
+};
+
+class sensorFusion : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        sensorFusion() {}
+        sensorFusion(const sensorFusion& other);
+        sensorFusion& operator =(const sensorFusion& other);
+};
+
+class waypointMgmt : public pathManagerState
```

Comment:
So rather than calling this the waypointMgmt state, let's call this the "cruising" state. Cuz Nixon is going to add a few more states in here that also use the waypoint manager module for landing.